analytics/agma/agma_module.go

```diff
@@ -65,8 +66,11 @@ func newAgmaLogger(cfg config.AgmaAnalytics, sender httpSender, clock clock.Cloc
 		maxEventCount:     int64(cfg.Buffers.EventCount),
 		maxDuration:       pDuration,
 		buffer:            buffer,
-		bufferCh:          make(chan []byte),
-		sigTermCh:         make(chan os.Signal, 1),
+		read: func(p []byte, b bytes.Buffer) (int, error) {
```

Comment:
I think it would be more flexible, and make more sense, to make the buffer settable (an input parameter) by the `newAgmaLogger()` call. For the test you can make a mock buffer (it would have to implement the full buffer interface, which is a little more work) with your reader method as the mock buffer's read method. This would also allow for the mock buffer to be used in other tests.